stdlib/2and3/socket.pyi

```diff
@@ -539,71 +589,101 @@ class socket:
     @overload
     def getsockopt(self, level: int, optname: int, buflen: int) -> bytes: ...
 
+    if sys.version_info >= (3, 7):
+        def getblocking(self) -> bool: ...
     def gettimeout(self) -> Optional[float]: ...
-    def ioctl(self, control: object,
-              option: Tuple[int, int, int]) -> None: ...
-    if sys.version_info < (3, 5):
-        def listen(self, backlog: int) -> None: ...
+
+    if sys.platform == 'win32':
+        def ioctl(self, control: object, option: Tuple[int, int, int]) -> None: ...
+
+    if sys.version_info >= (3, 5):
+        def listen(self, __backlog: int = ...) -> None: ...
     else:
-        def listen(self, backlog: int = ...) -> None: ...
-    # TODO the return value may be BinaryIO or TextIO, depending on mode
-    def makefile(self, mode: str = ..., buffering: int = ...,
-                 encoding: str = ..., errors: str = ...,
-                 newline: str = ...) -> Any:
-        ...
+        def listen(self, __backlog: int) -> None: ...
+    # TODO makefile's return value may be BinaryIO or TextIO, depending on mode
+    def makefile(self,
+                 mode: str = ...,
+                 buffering: int = ...,
+                 encoding: str = ...,
+                 errors: str = ...,
+                 newline: str = ...) -> Any: ...
```

Comment:
In Python 2, this only has two arguments, in Python 3 all other arguments are keyword-only. Also, all arguments except `mode` can be `None` in Python 3.